src/nanovg_pick.c

```diff
@@ -0,0 +1,1742 @@
+				if (ninflections)
+				{
+					nvg__smallsort(inflections, ninflections);
+
+					float previnfl = 0;
+					float t;
+					
+					float pointsA[8], pointsB[8];
+
+					int infl;
+					for (infl = 0; infl < ninflections; ++infl)
+					{
+						if (nvg__absf(inflections[infl] - previnfl) < NVG_PICK_EPS)
+							continue;
+							
+						t = (inflections[infl] - previnfl) * (1.0f / (1.0f - previnfl));
+						
+						previnfl = inflections[infl];
+							
+						nvg__splitBezier(points, t, pointsA, pointsB);
+
+						firstPoint = nvg__pickSubPathAddPoints(ps, psp, &pointsA[2], 3);
+						nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, (infl == 0) ? NVG_PICK_CORNER : 0);
+
+						memcpy(points, pointsB, sizeof(float) * 8);												
+					}
+					
+					firstPoint = nvg__pickSubPathAddPoints(ps, psp, &pointsB[2], 3);
+					nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, 0);
+				}
+				else
+				{
+					firstPoint = nvg__pickSubPathAddPoints(ps, psp, &commands[i+1], 3);
+					nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, NVG_PICK_CORNER);
+				}
+			}
+			i += 7;
+			break;
+		case NVG_CLOSE:
+			{
+				if (	ps->points[(ps->npoints - 1) * 2] != start[0]
+					||	ps->points[(ps->npoints - 1) * 2 + 1] != start[1])
+				{
+					firstPoint = nvg__pickSubPathAddPoints(ps, psp, start, 1);
+					nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, NVG_LINETO, NVG_PICK_CORNER);
+				}
+				psp->closed = 1;
+				
+				i++;
+			}	
+			break;
+		case NVG_WINDING:
+			psp->winding = (short)(int)commands[i+1];
+			if (psp->winding == NVG_HOLE)
+				hasHoles = 1;
+			i += 2;
+			break;
+		default:
+			i++;
+		}
+	}
+
+	NVGstate* state = nvg__getState(context);
+	
+	pp->flags = 	(hasHoles ? NVG_PICK_HASHOLES : 0)
+				| 	(forStroke ? NVG_PICK_STROKE : NVG_PICK_FILL);
+	pp->subPaths = psp;
+	pp->strokeWidth = state->strokeWidth * 0.5f;
+	pp->miterLimit = state->miterLimit;
+	pp->lineCap = (short)state->lineCap;
+	pp->lineJoin = (short)state->lineJoin;
+
+	float totalBounds[4];
+	nvg__initBounds(totalBounds);
+	NVGsegment* segments = NULL;
+	const NVGsegment* seg = NULL;
+	NVGpickSubPath *curpsp;
+	int s;
+
+	for (curpsp = psp; curpsp; curpsp = curpsp->next)
+	{	
+		if (forStroke)
+			nvg__pickSubPathAddStrokeSupports(ps, curpsp, pp->strokeWidth, pp->lineCap, pp->lineJoin, pp->miterLimit);
+		else
+			nvg__pickSubPathAddFillSupports(ps, curpsp);
+		
+		segments = &ps->segments[curpsp->firstSegment];
+		nvg__initBounds(curpsp->bounds);
+		for (s = 0; s < curpsp->nsegments; ++s)
+		{
+			seg = &segments[s];
+			NVG_PICK_DEBUG_BOUNDS(seg->bounds);		
+			nvg__unionBounds(curpsp->bounds, seg->bounds);
+		}
+
+		nvg__unionBounds(totalBounds, curpsp->bounds);
+	}
+	
+	// Scissor rect is handled by intersecting it with the paths bounds
+	// and setting a flag to control an additional check for picking.
+	if (state->scissor.extent[0] != -1.0f)
+	{	
+		float scissor[4];
+		
+		scissor[0] = state->scissor.xform[4] - state->scissor.extent[0];
```

Comment:
I think this is not correct. The scissor rect can become non-axis aligned. We should be able to mimic the code from the shader (distance to oriented rect), to calculate if in scissor. Let me know if you need more details.